app/controllers/device_controller.rb

```diff
@@ -43,7 +47,7 @@ def raw_data
   end
```

Comment:
I think I'd like to have handled that with validation on the records, to prevent an invalid device or report getting into the database in the first place. I think the app in general needs to be a little less generally fragile though, and this is likely something we'll look into next week (e.g. accessing a page for a user that doesn't exist takes you to an ugly rails error page) and I think this would go along with that.